examples/models/llama/model.py

```diff
@@ -171,7 +172,16 @@ def __init__(self, **kwargs):
         # Within the device="meta" context, tensors that are created do not carry data.
         # They possess all other metadata a tensor carries such as size, stride, requires_grad.
         with torch.device("meta"):
+            # Model itself is loaded in default dtype, fp32.
             self.model_ = Transformer(model_args)
+            if self.dtype:
+                self.model_.to(dtype=self.dtype)
+
+            # Convert the model's weights only to the checkpoint's dtype, so that
+            # the checkpoint can be loaded into the model's state dict in its
+            # own dtype w/o potential precision loss.
+            for param in self.model_.parameters():
+                param.data = param.data.to(dtype=self.checkpoint_dtype)
```

Comment:
We shouldnt have to do this if the checkpoint is directly loaded, no? Not sure whats happening with `self.model_.to(....`